file.go

```diff
@@ -206,6 +206,9 @@ func (f *File) readPageIndex(section *bufferedSectionReader, decoder *thrift.Dec
 	return columnIndexes, offsetIndexes, nil
 }
 
+// NumRows returns the number of rows in the file.
+func (f *File) NumRows() int64 { return f.metadata.NumRows }
```

Comment:
Does this need to be on `*File` or should it just be on `File`?

Reply:
Either way would work, for the sake of consistency with every other methods I'd keep the receiver be a `*File`, making a copy when passing by value doesn't really add much to readability in my opinion.